### PR TITLE
Updated spec text on encryption

### PR DESCRIPTION
**Pull request by thehenrytsai**
For #59.

### CODE REVIEW

File: spec/spec.md [hunk 1]
```diff
@@ -1602,16 +1558,54 @@ The `PermissionQuery` method exists to facilitate lookup of any retained Permiss
 
 The Sync interface and its methods allow different Decentralized Web Nodes to communicate and sync on the state of the data they contain, including replication of messages and files.
 
-## Supported Encryption Schemes
+## Encryption
+
+Each Decentralized Web Node (DWN) supports encryption at the individual message level. Encryption of data is performed using a symmetric key unique to each message. The symmetric key is then encrypted with one or more public keys derived using HMAC-based Key Derivation Function as specified per [[spec:RFC5869]] together with one of the [key derivation path schemes](#key-derivation-path-schemes) defined in this specification.  Each key derivation path scheme is optimized for a particular usage pattern of the Decentralized Web Node. Application and protocol developers can select schemes that best fit their requirements.
```
Comment (TallTed) [inline spec/spec.md:1563]:
```suggestion
Each Decentralized Web Node (DWN) supports encryption at the individual message level. Encryption of data is performed using a symmetric key unique to each message. The symmetric key is then encrypted with one or more public keys derived using the HMAC-based Key Derivation Function as specified in [[spec:RFC5869]] together with one of the [key derivation path schemes](#key-derivation-path-schemes) defined in this specification.  Each key derivation path scheme is optimized for a particular usage pattern of the Decentralized Web Node. Application and protocol developers can select schemes that best fit their requirements.
```

File: spec/spec.md [hunk 2]
```diff
@@ -1602,16 +1558,54 @@ The `PermissionQuery` method exists to facilitate lookup of any retained Permiss
...
+```
+
+
+#### Protocol Context Scheme
+
+This scheme allows the initiator of a new record context to derive and specify a public key that encrypts all symmetric keys within the context. The holder of the corresponding context derived private key can decrypt all messages within the context.
+
+Under this scheme, a recommended pattern for distributing the private key to a participant is to include the encrypted private key in the participant's _role record_ itself. This encryption uses the public key declared in the recipient's own instance of the configuration of the same protocol. This approach ensures that all essential cryptographic materials remain within the record context hierarchy.
+
+The derivation path segments of this scheme must the structure:
```
Comment (TallTed) [inline spec/spec.md:1591]:
```suggestion
The derivation path segments of this scheme must use the structure below:
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Editorial — grammar and such

File: spec/spec.md [hunk 3]
```diff
@@ -896,7 +833,21 @@ directory of the specification.
...
   - The object ****MAY**** contain a `published` property, and if present its value ****Must**** be a boolean indicating the record's publication state. A value of `true` indicates the record has been published for public queries and consumption without requiring authorization. A value of `false` or the absence of the property indicates the record ****MUST NOT**** be served in response to public queries that lack proper authorization.
-  - The object ****MAY**** contain an `encryption` property, and if present its value ****Must**** be a string that matches one of the [Supported Encryption Formats](#supported-encryption-format), indicating the encryption format with which the data is encrypted. The absence of this property indicates the data is not encrypted.
+
+  - The object ****MAY**** contain an `encryption` property. The object ****MUST**** contain the `encryption` property if the data is encrypted. The absence of this property indicates the data is not encrypted. If present its value ****Must**** be a JSON object composed as follows:
```
[the remark this answers is not in the document]
Comment (andorsk) [reply spec/spec.md:837]: @thehenrytsai can we merge these in. Thank you @TallTed for the comments

**Review by andorsk — APPROVED** [review]
first draft looks good. merging in and we'll come back to this later after we've gotten the rest of the sections in better order.

**Pull request merged**